src/workerd/server/container-client.c++

```diff
@@ -286,17 +286,22 @@ kj::Promise<void> ContainerClient::createContainer(
   // where the container we're managing is stuck at "exited" state.
   hostConfig.initRestartPolicy().setName("on-failure");
 
-  auto endpoint = kj::str("/containers/create?name=", containerName);
-  auto response = co_await dockerApiRequest(
-      network, kj::str(dockerPath), kj::HttpMethod::POST, kj::mv(endpoint), codec.encode(jsonRoot));
+  auto response = co_await dockerApiRequest(network, kj::str(dockerPath), kj::HttpMethod::POST,
+      kj::str("/containers/create?name=", containerName), codec.encode(jsonRoot));
 
-  // statusCode 201 refers to "container created successfully"
   // statusCode 409 refers to "conflict". Occurs when a container with the given name exists.
-  // Both are fine, so long as the container exists. Though we might want to call destroy and
-  // recreate on 409 in the future.
+  // In that case we destroy and re-create the container.
+  if (response.statusCode == 409) {
+    co_await destroyContainer();
+    response = co_await dockerApiRequest(network, kj::str(dockerPath), kj::HttpMethod::POST,
+        kj::str("/containers/create?name=", containerName), codec.encode(jsonRoot));
+  }
+
+  // statusCode 201 refers to "container created successfully"
   if (response.statusCode != 201 && response.statusCode != 409) {
```

Comment:
Should `!= 409` still be there?

Reply:
> I assume that's a bug with docker that we can ignore.

I was under the same impression (i.e. it should not happen) but I had a different conclusion: may we should `JSG_REQUIRE(response.statusCode != 409)` ?